Admin/CrossTenantMailboxMigrationValidation.ps1

```diff
@@ -216,6 +232,23 @@ function CheckObjects {
                     Write-Verbose -Message "Mailbox is not under any Organizational Hold"
                 }
 
+                #Verify if SOURCE mailbox has an Archive, and if it does, check if there's any item within recoverable items SubstrateHolds folder.
+                if ($SourceObject.ArchiveGUID -notmatch "00000000-0000-0000-0000-000000000000") {
+                    Write-Verbose -Message "Informational: SOURCE mailbox has an Archive enabled, checking if there's any SubstrateHold folder present"
+                    if ((Get-SourceMailboxFolderStatistics $SourceObject.ArchiveGuid -FolderScope RecoverableItems | Where-Object { $_.Name -eq 'SubstrateHolds' })) {
+                        Write-Verbose -Message "Informational: SubstrateHolds folder found in SOURCE Archive mailbox, checking if there's any content inside it"
+                        if ((Get-SourceMailboxFolderStatistics $SourceObject.ArchiveGuid -FolderScope RecoverableItems | Where-Object { $_.Name -eq 'SubstrateHolds' }).ItemsInFolder -gt 0) {
```

Comment:
Let's avoid running the same cmdlet twice with different filters. 

Store the cmdlet to a variable, then in the `if` statement do the statement to determine if we want to enter.

Reply:
thx for the tip. 
Changed it to as advised